xa/java/jms/department-helidon-teq/src/main/java/com/oracle/mtm/sample/Configuration.java

```diff
@@ -0,0 +1,166 @@
+Software is contributed by such licensors), **
+without restriction, including without limitation the rights to copy, create derivative works of, display, perform, and distribute the Software and make, use, sell,
+offer for sale, import, export, have made, and have sold the Software and the Larger Work(s), and to sublicense the foregoing rights on either these or other terms. **
+
+This license is subject to the following condition: The above copyright notice and either this complete permission notice or at a minimum a reference to the UPL must be
+included in all copies or substantial portions of the Software. **
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+package com.oracle.mtm.sample;
+
+import oracle.jakarta.jms.AQjmsFactory;
+import oracle.jakarta.jms.AQjmsSession;
+import oracle.jakarta.jms.AQjmsTextMessage;
+import oracle.jakarta.jms.AQjmsTopicSubscriber;
+import oracle.tmm.common.TrmConfig;
+import oracle.tmm.jta.common.DataSourceInfo;
+import oracle.ucp.jdbc.PoolDataSource;
+import oracle.ucp.jdbc.PoolDataSourceFactory;
+import oracle.ucp.jdbc.PoolXADataSource;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.context.Initialized;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import jakarta.jms.JMSException;
+import jakarta.jms.Session;
+import jakarta.jms.Topic;
+import jakarta.jms.TopicConnection;
+import jakarta.jms.TopicSession;
+import jakarta.jms.XATopicConnectionFactory;
+import java.lang.invoke.MethodHandles;
+import java.sql.SQLException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@ApplicationScoped
+public class Configuration {
+
+    private PoolXADataSource xaDataSource;
+    private PoolDataSource dataSource;
+    private Logger logger;
+
+    @Inject
+    @ConfigProperty(name = "departmentDataSource.url")
+    String url;
+    @Inject
+    @ConfigProperty(name = "departmentDataSource.user")
+    String user;
+    @Inject
+    @ConfigProperty(name = "departmentDataSource.password")
+    String password;
+
+    @Inject
+    @ConfigProperty(name = "departmentDataSource.rmid")
+    String rmid;
+
+    private void init(@Observes @Initialized(ApplicationScoped.class) Object event) {
+        initialiseTrmLogger();
+        initialiseDataSource();
+        new Thread(() -> {
+            consumeEvents();
+        }).start();
+
+    }
+
+    private void consumeEvents() {
```

Comment:
a simple comment may need here for this daemon consumer

Reply:
Done